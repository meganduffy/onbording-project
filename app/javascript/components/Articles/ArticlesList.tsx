import React, { useState, useEffect } from "react";
import ArticlesDetails from './ArticlesDetails';
import { array } from "prop-types";

export interface ArticlesListProps {
    link?:string
}

export default function ArticlesList(props) {

    let display = (<h3>It looks as though there's nothing here yet!</h3>)

    function formatDate(date) {
        let msec = Date.parse(date)
        let newDate = new Date(msec)
        return newDate.toDateString()
    }

    if (props.articles.length >= 1) {
        display = <div>
            {props.articles.map((article, index) => 
                <ArticlesDetails 
                key={article.id} 
                title={article.title} 
                content={article.content.substring(0, 300) + "..."}
                firstName={article.user.first_name}
                created_at={formatDate(article.created_at)}
                link={`/articles/${article.id}`}
                linkText="View Full Article" />
            )}
        </div> 
    } 

    return (
        <div>
            {display}
        </div>      
    )
}
