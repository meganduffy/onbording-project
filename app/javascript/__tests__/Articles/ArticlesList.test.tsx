import React from 'react';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import renderer from 'react-test-renderer';
import ArticlesList from '../../components/Articles/ArticlesList';

let container = null;

let singleArticle = [{id: 1, title: "Title1", content: "Lorem Ipsum blah blah", 
created_at: "2020-03-27T10:53:09.881Z", user: {first_name: "Megan"}}]

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders correctly', ()=>{
  const tree = renderer.create(<ArticlesList 
    articles={singleArticle} />).toJSON();
  expect(tree).toMatchSnapshot();
});
