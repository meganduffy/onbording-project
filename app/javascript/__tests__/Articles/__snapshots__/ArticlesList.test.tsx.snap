// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList Snapshots should render the Article List when logged in 1`] = `
<div>
  <div>
    <div>
      <div
        class="outer-article-container"
        data-testid="article-details"
      >
        <div
          class="articles-details"
        >
          <h2
            class="articles-title"
          >
            mechanism momentum museum minimum mushroom
          </h2>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillu...
          </p>
          <div
            class="articles-details-info"
          >
            <p>
              Megan
               
            </p>
            <p>
              Fri Mar 27 2020
            </p>
          </div>
        </div>
        <div
          class="articles-aside"
        >
          <a
            href="/articles/1"
          >
            View Full Article
          </a>
          <a
            href="/articles/1/edit"
          >
            Edit Article
          </a>
          <a
            href="/articles/1/delete"
          >
            Delete Article
          </a>
        </div>
      </div>
      <div
        class="outer-article-container"
        data-testid="article-details"
      >
        <div
          class="articles-details"
        >
          <h2
            class="articles-title"
          >
            green vegetable saw comparison
          </h2>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillu...
          </p>
          <div
            class="articles-details-info"
          >
            <p>
              Matthew
               
            </p>
            <p>
              Sun Mar 29 2020
            </p>
          </div>
        </div>
        <div
          class="articles-aside"
        >
          <a
            href="/articles/2"
          >
            View Full Article
          </a>
          <a
            href="/articles/2/edit"
          >
            Edit Article
          </a>
          <a
            href="/articles/2/delete"
          >
            Delete Article
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ArticleList Snapshots should render the Article List when logged out 1`] = `
<div>
  <div>
    <div>
      <div
        class="outer-article-container"
        data-testid="article-details"
      >
        <div
          class="articles-details"
        >
          <h2
            class="articles-title"
          >
            mechanism momentum museum minimum mushroom
          </h2>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillu...
          </p>
          <div
            class="articles-details-info"
          >
            <p>
              Megan
               
            </p>
            <p>
              Fri Mar 27 2020
            </p>
          </div>
        </div>
        <div
          class="articles-aside"
        >
          <a
            href="/articles/1"
          >
            View Full Article
          </a>
        </div>
      </div>
      <div
        class="outer-article-container"
        data-testid="article-details"
      >
        <div
          class="articles-details"
        >
          <h2
            class="articles-title"
          >
            green vegetable saw comparison
          </h2>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillu...
          </p>
          <div
            class="articles-details-info"
          >
            <p>
              Matthew
               
            </p>
            <p>
              Sun Mar 29 2020
            </p>
          </div>
        </div>
        <div
          class="articles-aside"
        >
          <a
            href="/articles/2"
          >
            View Full Article
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ArticleList Snapshots should render the Article List when there is only one article 1`] = `
<div>
  <div>
    <div>
      <div
        class="outer-article-container"
        data-testid="article-details"
      >
        <div
          class="articles-details"
        >
          <h2
            class="articles-title"
          >
            mechanism momentum museum minimum mushroom
          </h2>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillu...
          </p>
          <div
            class="articles-details-info"
          >
            <p>
              Megan
               
            </p>
            <p>
              Fri Mar 27 2020
            </p>
          </div>
        </div>
        <div
          class="articles-aside"
        >
          <a
            href="/articles/1"
          >
            View Full Article
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
